packages/integration/lib/src/bloc/universal_feed/universal_feed_bloc.dart

```diff
@@ -60,28 +59,47 @@ class LMFeedBloc extends Bloc<LMFeedEvent, LMFeedState> {
           .map((e) => LMTopicViewDataConvertor.toTopic(e))
           .toList();
     }
-    GetFeedResponse? response = await LMFeedCore.instance.lmFeedClient.getFeed(
+    GetFeedResponse response = await LMFeedCore.instance.lmFeedClient.getFeed(
       (GetFeedRequestBuilder()
             ..page(event.offset)
             ..topics(selectedTopics)
             ..pageSize(10))
           .build(),
     );
 
-    if (response == null) {
+    if (!response.success) {
       emit(const LMFeedUniversalFeedErrorState(
           message: "An error occurred, please check your network connection"));
     } else {
-      users.addAll(response.users.map((key, value) =>
-          MapEntry(key, LMUserViewDataConvertor.fromUser(value))));
-      topics.addAll(response.topics.map((key, value) =>
-          MapEntry(key, LMTopicViewDataConvertor.fromTopic(value))));
-
+      if (response.users != null) {
+        users.addAll(response.users!.map((key, value) =>
+            MapEntry(key, LMUserViewDataConvertor.fromUser(value))));
+      }
+      if (response.topics != null) {
+        topics.addAll(response.topics!.map((key, value) =>
+            MapEntry(key, LMTopicViewDataConvertor.fromTopic(value))));
+      }
+      if (response.widgets != null) {
+        widgets.addAll(response.widgets!.map((key, value) =>
+            MapEntry(key, LMWidgetViewDataConvertor.fromWidgetModel(value))));
+      }
+      if (response.posts == null) {
+        emit(const LMFeedUniversalFeedErrorState(
```

Comment:
error message should be returned from backend if that is empty then only we'll provide our own